### PR TITLE
Re-add get_forward_progress_guarantee, this time with spec.

### LINKED ISSUE #83: Add design and wording for `get_forward_progress_guarantee`
@griwes imagines this as a scheduler query that returns an enum value that describes the forward progress guarantee.

### PR DESCRIPTION
**Pull request by griwes**
Resolves #83

### CODE REVIEW

File: std_execution.bs [hunk 1]
```diff
@@ -553,6 +553,17 @@ merged into a single one in an asynchronous fashion.
 `transfer_when_all` accepts an additional scheduler argument. It returns a sender whose value [=completion scheduler=] is the scheduler provided as an argument, but otherwise behaves the same as `when_all`. You can think of it as a composition of
 `transfer(when_all(inputs...), scheduler)`, but one that allows for better efficiency through customization.
 
+## Schedulers advertise their forward progress guarantees ## {#design-fpg}
+
+To decide whether a scheduler (and its associated execution context) is sufficient for a specific task, it may often be necessary to know what kind of forward progress guarantees it provides for the execution agents it creates. The C++ Standard defines the following
```
Comment (brycelelbach) [inline std_execution.bs:558]: "may often be necessary" -> "may be necessary"

File: std_execution.bs [hunk 2]
```diff
@@ -553,6 +553,17 @@ merged into a single one in an asynchronous fashion.
...
+forward progress guarantees:
+
+* <i>concurrent</i>, which requires that a thread makes progress <i>eventually</i>;
+* <i>parallel</i>, which requires that a thread makes progress once it executes a step; and
+* <i>weakly parallel</i>, which does not require that the thread makes progress.
```
Comment (brycelelbach) [inline std_execution.bs:563]: Check with @ogiroux and make sure these words don't offend his sensibilities.

File: std_execution.bs [hunk 3]
```diff
@@ -553,6 +553,17 @@ merged into a single one in an asynchronous fashion.
...
+
+This paper introduces a scheduler query function, `get_forward_progress_guarantee`, which returns one of values of a new `enum` type, `forward_progress_guarantee`, which matches one of the gurantees defined by the C++ Standard.
```
Comment (brycelelbach) [inline std_execution.bs:565]: "one of values of a new `enum` type" -> "one of the enumerators of a new `enum` type".

Comment (brycelelbach) [reply std_execution.bs:565]: "`forward_progress_guarantee`, which matches one of the guarantees defined by the C++ Standard." -> "`forward_progress_gaurantee`. Each enumerator of `forward_progress_guarantee` corresponds to one of the aforementioned guarantees."

(Split it into two sentences, and reword the second one)

File: std_execution.bs [hunk 4]
```diff
@@ -1544,6 +1555,13 @@ namespace std::execution {
   template&lt;class S>
     concept scheduler = <i>see-below</i>;
 
+  // [execution.schedulers.queries], scheduler queries
+  enum class forward_progress_guarantee;
+  inline namespace <i>unspecified</i> {
```
Comment (brycelelbach) [inline std_execution.bs:1560]: Why an `inline namespace`?

File: std_execution.bs [hunk 5]
```diff
@@ -1755,6 +1773,30 @@ namespace std::execution {
...
+
+<pre highlight=c++>
+enum class forward_progress_guarantee {
+    concurrent,
+    parallel,
+    weakly_parallel
+};
+</pre>
+
+1. `execution:;get_forward_progress_guarantee` is used to ask a scheduler about the forward progress guarantees of execution agents created by that scheduler.
```
Comment (brycelelbach) [inline std_execution.bs:1788]: `execution:;get_forward_progress_guarantee` -> `execution::get_forward_progress_guarantee`.

**Review by brycelelbach — APPROVED** [review]
Looks good to merge pending resolution of comments.

File: std_execution.bs (hunk 4 above)
Comment (griwes) [reply std_execution.bs:1560]: That's how all CPOs are wrapped.

File: std_execution.bs (hunk 2 above)
Comment (griwes) [reply std_execution.bs:565]: Olivier says it's good.

**Pull request merged**